src/components/atoms/SvgLoop/SvgLoop.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef, useState } from "react";
+import classNames from "classnames";
+
+export interface SvgLoopProps
+  extends React.ImgHTMLAttributes<HTMLImageElement> {
+  sources: string[];
+  delay?: number;
+}
+
+const pause: (delay: number) => Promise<void> = (delay) =>
+  new Promise((resolve) => setTimeout(resolve, delay));
+
+export const SvgLoop: React.FC<SvgLoopProps> = ({
+  sources,
+  delay = 1000,
+  alt,
+  ...extraProps
+}) => {
+  const ref = useRef<HTMLImageElement>(null);
+  const [timeout, setTimeout] = useState(delay);
+  const [index, setIndex] = useState(0);
+
+  useEffect(() => {
+    (async () => {
```

Comment:
You can use `useAsync` from `react-use`

Reply:
Was the first thing I tried. It didn't call the cleanup function like `useEffect` does. E.g.

```ts
     return () => {
      console.log("CLEANUP CALLED");
      setTimeout(0);
    };
```
![sparkle-cleanup-01](https://user-images.githubusercontent.com/79229621/130445501-a2f43584-6b25-426f-9152-262c7b4f2250.png)